stablehlo/conversions/linalg/transforms/StablehloLegalizeToLinalg.cpp

```diff
@@ -419,29 +419,49 @@ struct BroadcastConverter final
   }
 };
 
+template <typename OpTy>
+int64_t getBroadcastSizes(OpTy op) {
+  if constexpr (std::is_same<OpTy, BroadcastOp>::value)
+    return op.getBroadcastSizes().size();
+  if constexpr (std::is_same<OpTy, BroadcastInDimOp>::value)
+    return op.getType().getRank() - op.getBroadcastDimensions().size();
+  std::string str;
+  llvm::raw_string_ostream os(str);
+  os << "getBroadcastSizes called on non-broadcast op: ";
+  op.print(os);
+  llvm::report_fatal_error(str.c_str());
+}
+
+template <typename OpTy, typename OpAdaptor>
+LogicalResult lowerBroadcastOp(ConversionPatternRewriter &rewriter,
+                               const TypeConverter *typeConverter, OpTy op,
+                               OpAdaptor adaptor) {
```

Comment:
I think you can use `OpTy::Adaptor` instead of taking a second template argument?

Reply:
Yep, good call! I'll apply that to the dot lowering as well (in a separate PR).